src/common/local_file_system.cpp

```diff
@@ -52,32 +52,50 @@ static void AssertValidFileFlags(uint8_t flags) {
 }
 
 #ifndef _WIN32
-bool LocalFileSystem::FileExists(const string &filename) {
-	if (!filename.empty()) {
-		if (access(filename.c_str(), 0) == 0) {
-			struct stat status;
-			stat(filename.c_str(), &status);
-			if (S_ISREG(status.st_mode)) {
-				return true;
-			}
+
+static FileType GetFileTypeWithAccessCheck(const std::string &filename) {
+	if (filename.empty()) {
+		return FileType::FILE_TYPE_INVALID;
```

Comment:
Is INVALID overloaded? Should we have a NONE type?

Reply:
Yeah a separate type would be cleaner

Reply:
++